python/pyspark/testing/pandasutils.py

```diff
@@ -341,6 +341,11 @@ def assertPandasOnSparkEqual(
 
     .. versionadded:: 3.5.0
 
+    .. deprecated:: 3.5.1
+        `assertPandasOnSparkEqual` will be removed in Spark 4.0.0.
+        Use `ps.testing.assert_frame_equal`, `ps.testing.assert_series_equal`
+        and `ps.testing.assert_index_equal` instead.
```

Comment:
Seems like these functions don't exist `branch-3.5`. Will they be backported?

Reply:
@itholic let's remove this in branch-3.5.

Reply:
Thanks for catching it out. Just created followup here: https://github.com/apache/spark/pull/43459.